evalkit_api_client/evalkit_api_client/v1_client.py

```diff
@@ -0,0 +1,178 @@
+        if not response.ok:
+            logger.debug('Error status code for url "{}"'.format(response.url))
+        if exit_on_error:
+            response.raise_for_status()
+
+        return response
+
+    def _get(self, *args, **kwargs) -> Response:
+        """
+        Sends a GET request to the API.
+        """
+        return self._send_request(
+            requests.get, *args, **kwargs)  # type: ignore
+
+    def _delete(self, *args, **kwargs) -> Response:
+        """
+        Sends a DELETE request to the API.
+        """
+        return self._send_request(
+            requests.delete, *args, **kwargs)  # type: ignore
+
+    def _post(self, *args, **kwargs) -> Response:
+        """
+        Sends a POST request to the API.
+        """
+        return self._send_request(
+            requests.post, *args, **kwargs)  # type: ignore
+
+    def _put(self, *args, **kwargs) -> Response:
+        """
+        Sends a PUT request to the API.
+        """
+        return self._send_request(
+            requests.put, *args, **kwargs)  # type: ignore
+
+    def _get_validated_page_json(self,
+                       url: str,
+                       headers: RequestHeaders = None,
+                       params: RequestParams = None) -> Dict:
+        response = self._get(url, headers=headers, params=params)
+
+        resp_json = response.json()
+        try:
+            page = resp_json['page']
+            page_size = resp_json['pageSize']
+            result_list = resp_json['resultList']
+        except KeyError:
+            raise APIPaginationException(
+                'Pagination is not available for this endpoint')
+
+        return resp_json
+
+    def _get_paginated(self,
+                       url: str,
+                       headers: RequestHeaders = None,
+                       params: RequestParams = None) -> Iterator[Dict]:
+        """
+        Send an API call to the evalkit server with pagination.
+
+        Returns a generator of dictionary objects.
+
+        TODO: Evalkit's pagination is broken, but this logic works the way
+        the Evalkit docs say it should.
+        """
+        if params is None:
+            params = {}
+
+        counter = 0
+        retrieve_next_page = True
+
+        while retrieve_next_page:
+            params = {'page': counter + 1}
+            resp_json = self._get_validated_page_json(url,
+                                                      headers=headers,
+                                                      params=params)
+
+            yield resp_json
+
+            if len(resp_json['resultList']) < resp_json['pageSize']:
+                retrieve_next_page = False
+            else:
+                counter += 1
+
+    def get_projects(self,
+                    params: RequestParams = None
+                    ) -> Iterator[Dict]:
+        """
+        Get the projects for a given account.
+
+        TODO: use pagination when available.
```

Comment:
Can you write this so that it would work with pagination as documented, or with the current not-actually-paginated results?

Reply:
Yup, it's actually already been written this way! You can read the tests to see it

Reply:
Great!